### DISCUSSION TITLE
DataTable: Row Selection is Highlighting Everything

### DISCUSSION THREAD

**Discussion by Jeff-Hamby** [primefaces/community#1457] [PrimeVue]
I need to use the PrimeVue Datatable Single Row Selection.  When a row is selected all the rows display on the control as highlighted instead of a single record.  However, when checking the data, only a single record is recorded.  

I'm using the Theme 
import 'primevue/resources/themes/lara-light-green/theme.css';

<DataTable 
v-model:selection="selectedAssignNot"
:value="districtListNotAssigned"
selectionMode="single"
 dataKey="id" 
 :metaKeySelection="false" 
@rowSelect="onRowSelectNot" 
@rowUnselect="onRowUnselectNot"
>
<Column field="USD" label="USD" />
Column field="USD_Name" label="Name" />
</DataTable>

"primeflex": "^3.3.1",
"primeicons": "^6.0.1",
"primevue": "^3.50.0",
 "vue": "^3.2.45",

**Comment by tugcekucukoglu** [comment]
Cannot replicate the problem. https://primevue.org/datatable/#single_row_selection

**Comment by joelogna** [comment]
If your dataKey isn't a property on your model you'll get this behavior. So in your case, if "id" isn't a property of the "districtListNotAssigned" object then it won't find the key and will just select all rows instead. At least that is what I spent an hour today figuring out in my code.
